Master ctauto294

jrdavisllbean opened the pull request:
Didier and Eder, the code in this branch is for CTAUTO-294. Please review when you have a free moment and let me know if I need to change anything.  In this class I ported the BaseMobile method named waitUntilLoaderImageDisappear(). I found this method helped when waiting for the BillingAddress page. Would it be possible to add to BaseWeb/BasePageObject.

    // TODO -- Possibly add to BasePageObject or BaseWEB -- JRD 9/23/16
    void waitUntilLoaderImageDisappear() {
        By loadingSpinningImage = By.id("ajaxWait");
        wait.until(ExpectedConditions.invisibilityOfElementLocated(loadingSpinningImage));
        this.waitForJQueryToComplete();
        this.waitForJavaScriptToComplete();
    }

didierllbean commented on src/main/java/com/llbean/automation/web/pages/desktopNTablet/ShippingAddress.java:
```diff
@@ -0,0 +1,181 @@
+        shippingAddress1.sendKeys(address1,Keys.TAB);
+        shippingAddress2.sendKeys(address2,Keys.TAB);
+        shippingCity.sendKeys(city,Keys.TAB);
+        new Select(shippingState).selectByValue(stateAbbreviation.toUpperCase());
```
Would you create this a global method in BasePageObject to handle any other similar implementations.
Similar to the code in BasePageObject class at line 107, method **selectItem**

didierllbean commented on src/main/java/com/llbean/automation/web/pages/mobile/ShippingAddressMobile.java:
```diff
@@ -0,0 +1,138 @@
+    void clickGotoBillingAddressScreenButton() {
+        continueToBillingAddress.click();
+        waitUntilLoaderImageDisappear();
+        wait.until(ExpectedConditions.visibilityOfElementLocated(billingAddressPresenceLocator));
```
Please create this as a global wait at the BasePageObject to use in future implementations

didierllbean commented:
I'm agree with your proposal of adding a waitUntilLoaderImageDisappear in web. I think the best place to put this is in the BaseWeb Class so using the same implementation that is on BaseMobile

jrdavisllbean commented:
Added selectItem(String textToSelect, WebElement selectField) to BasePageObject, waitUntilLoaderImageDisappear() to BaseWeb, implemented feedback in the Shipping Address classes

didierllbean commented on src/main/java/com/llbean/automation/web/pages/desktopNTablet/BaseWeb.java:
```diff
@@ -18,4 +22,12 @@ public BaseWeb(WebDriver driver) {
+
+    /*This wait until the loading image disapear from the page*/
+    public void waitUntilLoaderImageDisappear(){
+        WebDriverWait wait = new WebDriverWait(this.getDriver(), 15);
```
Hi @jrdavisllbean  you can remove this line and just use wait. due to there is a variable wait in the BasePageObject, implementing a Fluent wait so this declaration is not needed
